examples/RED/main.go

```diff
@@ -34,7 +34,7 @@ func main() {
 	})
 
 	// dereference struct to avoid panic
-	err = promstrap.RegisterMetrics(*reporter)
+	err = promstrap.RegisterComplexMetrics(*reporter)
```

Comment:
So this really is an opinion thing but I think it's better not to pass things by reference (unless we are mutating something in the ``Register`` method) in case someone forgets to do nil reference checks. It's often the cause of a lot of bugs. Passing in the entire struct doesn't decrease performance by much unless you have one really big massive struct. Someone on my last team did the performance check to verify.